src/scripts.js

```diff
@@ -248,13 +283,43 @@ function handleNavButtons(event) {
     pageTitleText.innerText = 'Whats Cookin';
   } else if (event.target.innerText === 'My Favorites') {
     displayFavoriteRecipesView();
+  } else if (event.target.innerText === 'My Pantry') {
+    displayPantryView();
   }
 }
 
+function displayPantryView() {
+  allRecipesView.classList.add('hidden');
+  singleRecipeView.classList.add('hidden');
+  pantryView.classList.remove('hidden');
+  displayUserPantry();
+  pageTitleText.innerText = 'My Pantry';
+}
+
 function displayFavoriteRecipesView() {
   allRecipesView.classList.remove('hidden');
+  pantryView.classList.add('hidden');
   singleRecipeView.classList.add('hidden');
   displayFavoriteRecipeCards({ recipes: currentUser.favoriteRecipes });
   pageTitleText.innerText = 'Favorite Recipes';
   searchInput.placeholder = 'Search Favorite Recipes';
+}
+
+function displayUserPantry() {
+  findPantryIngredientNames();
+  pantryList.innerHTML = '';
+  currentUser.pantry.pantry.forEach(ingredient => {
+    pantryList.innerHTML +=
+      `<li class="single-recipe-info">
+    <p class="single-recipe-number">${ingredient.amount}</p>
+    <p class="single-recipe-ingredient">${ingredient.name}</p>
+     </li>`
+  })
+}
+
+function findPantryIngredientNames() {
+  currentUser.pantry.pantry.map(ingredient => {
+    return ingredient.name =
+      ingredientRepo.returnIngredientName(ingredient.ingredient)
```

Comment:
Possibly a rename here, since this is adding ingredient names to a user's pantry. Maybe we call it, addIngredientNames?

Reply:
Yeah I like it! Updated the function name, and pushed it up!